dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/consumer/TaskPriorityQueueConsumer.java

```diff
@@ -120,7 +121,7 @@ public void run() {
      * batch dispatch with thread pool
      */
     private List<TaskPriority> batchDispatch(int fetchTaskNum) throws TaskPriorityQueueException, InterruptedException {
-        List<TaskPriority> failedDispatchTasks = new ArrayList<>();
+        List<TaskPriority> failedDispatchTasks = Collections.synchronizedList(new ArrayList<>());
```

Comment:
This is the funtion execute in the single thread, so is it necessary to use the `Collections.synchronizedList`? 
Will it affect performance?

Reply:
In this method, the thread pool consumerThreadPoolExecutor operates on the variable failedDispatchTasks collection.

![image](https://user-images.githubusercontent.com/37063904/156955399-edf2a5ae-e7c5-43bc-b1a2-7717a89d0753.png)

Reply:
Yes, this list is used in multiple threads, it's needed to use threadsafe collection here.